File: programmers/01.hash/03.전화번호 목록/python/other2.py
def solution(phoneBook):
    phoneBook.sort()

    for p1, p2 in zip(phoneBook, phoneBook[1:]):
        print(p1,p2)
        if p2.startswith(p1):
            return False
    return True



# test
print(solution([ "97674223", "1195524421","119"]))
# print(solution(["123","456","789"]))
# print(solution(["12","123","1235","567","88"]))

File: programmers/01.알고리즘/01.hash/05.베스트앨범/js/other2.js
function solution(genres, plays) {
    const count = {};
    let answer = [];
    const acc = genres.reduce((a, c, i) => {
        count[c] ? count[c].push([i, plays[i]]) : (count[c] = [[i, plays[i]]]);
        return a.set(c, a.get(c) ? a.get(c) + plays[i] : plays[i]), a;
    }, new Map());
    console.log(...acc);

    [...acc]
        .sort((a, b) => b[1] - a[1])
        .map((v) => {
            answer = answer.concat(count[v[0]].sort((c, d) => d[1] - c[1]).slice(0, 2));
        });
    return answer.map((v) => v[0]);
}

// test
console.log(
    solution(["classic", "pop", "classic", "classic", "pop"], [500, 600, 150, 800, 2500])
);
